lib/puppet/provider/service/init.rb

```diff
@@ -152,6 +152,15 @@ def restartcmd
     (@resource[:hasrestart] == :true) && [initscript, :restart]
   end
 
+  if Facter.value(:osfamily) == "Solaris"
+    def texecute(type, command, fof = true, squelch = false, combine = true)
+      if type =~ /^start/
+        command =  ["/usr/bin/ctrun -l none", command].flatten.join(" ")
+      end
+      super(type, command, fof, squelch, combine)
+    end
+  end
+
```

Comment:
I wouldn't conditionally override the `texecute` method as it leads to subtle issues, e.g. the call to `Facter.value` will execute on the master when it loads the provider (in the ruby `require` sense). It also makes it hard to test because you can't stub the behavior. It'd be better to always override the `texecute` method, and conditionally wrap the command if it's Solaris and type starts with `start`.